### PR TITLE
[NUI] Make capture api public

### PR DESCRIPTION
**Pull request by bshsqa**
- This patch open some API as public
 - Developer can use some Start method to start capture.
 - And also the developer can use finished signal.

Signed-off-by: seungho <sbsh.baek@samsung.com>

### API Changes
https://code.sec.samsung.net/jira/browse/TCSACR-414

### CODE REVIEW

**Review by hyunjushin — APPROVED** [review]
2020->2021

File: src/Tizen.NUI/src/public/Utility/Capture.cs [hunk 1]
```diff
@@ -65,32 +67,75 @@ internal Capture(IntPtr cPtr, bool cMemoryOwn) : base(cPtr, cMemoryOwn)
...
+        /// <summary>
+        /// Size of capturing region
+        /// Size value for each direction is required larger than zero.
+        /// Default is window size.
+        /// </summary>
+        /// <since_tizen> 9 </since_tizen>
+        public Size Size { get; set; } = new Size(0, 0, 0);
+
+        /// <summary>
+        /// image file path to be saved as a file.
```
Comment (Seoyeon2Kim) [inline src/Tizen.NUI/src/public/Utility/Capture.cs:85]: Please use a capital letter for the first word.

File: src/Tizen.NUI/src/public/Utility/Capture.cs [hunk 2]
```diff
@@ -65,32 +67,75 @@ internal Capture(IntPtr cPtr, bool cMemoryOwn) : base(cPtr, cMemoryOwn)
...
+        /// If path is empty string, the captured result is not be saved as a file.
```
Comment (Seoyeon2Kim) [inline src/Tizen.NUI/src/public/Utility/Capture.cs:86]: 'is not be saved' this sentence uses duplicated verbs,, :)

```suggestion
        /// If path is empty string, the captured result is not saved as a file.
```

File: src/Tizen.NUI/src/public/Utility/Capture.cs [hunk 3]
```diff
@@ -65,32 +67,75 @@ internal Capture(IntPtr cPtr, bool cMemoryOwn) : base(cPtr, cMemoryOwn)
...
+        /// </summary>
+        /// <since_tizen> 9 </since_tizen>
+        public string Path { get; set; } = null;
+
+        /// <summary>
+        /// background color of captured scene
```
Comment (Seoyeon2Kim) [inline src/Tizen.NUI/src/public/Utility/Capture.cs:92]:
```suggestion
        /// Background color of captured scene.
```

File: src/Tizen.NUI/src/public/Utility/Capture.cs [hunk 4]
```diff
@@ -65,32 +67,75 @@ internal Capture(IntPtr cPtr, bool cMemoryOwn) : base(cPtr, cMemoryOwn)
...
-            base.Dispose(type);
         }
 
-        /// This will not be public opened.
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        protected override void ReleaseSwigCPtr(HandleRef swigCPtr)
+        /// <summary>
+        /// Start capture and save the image as a file.
+        /// </summary>
+        /// <param name="source">source View or Layer to be used for capture.
```
Comment (Seoyeon2Kim) [inline src/Tizen.NUI/src/public/Utility/Capture.cs:119]: Please add </param> at the end of this sentence.

File: src/Tizen.NUI/src/public/Utility/Capture.cs [hunk 5]
```diff
@@ -65,32 +67,75 @@ internal Capture(IntPtr cPtr, bool cMemoryOwn) : base(cPtr, cMemoryOwn)
...
+                size = NUIApplication.GetDefaultWindow().Size;
+            }
+            if (null == path)
+            {
+                path = "";
+            }
+            if (source is View || source is Layer)
+            {
+                Interop.Capture.Start4(SwigCPtr, source.SwigCPtr, Position.SwigCPtr, new Vector2(size.Width, size.Height).SwigCPtr, path, new Vector4(ClearColor.R, ClearColor.G, ClearColor.B, ClearColor.A).SwigCPtr);
+
```
Comment (Seoyeon2Kim) [inline src/Tizen.NUI/src/public/Utility/Capture.cs:136]: Please remove the build warnings.
-> Call System.IDisposable.Dispose on object created by 'new Vector2(size.Width, size.Height)' before all references to it are out of scope.

You can define Vector2 and Vector4 instances separately.

File: src/Tizen.NUI/src/public/Utility/Capture.cs [hunk 6]
```diff
@@ -333,12 +378,39 @@ public NativeImageSource GetNativeImageSource()
             Tizen.Log.Debug("NUI", $"GetNativeImageSource()");
             return new NativeImageSource(Interop.Capture.GetNativeImageSourcePtr(SwigCPtr), true);
         }
+
+        /// <summary>
+        /// Dispose
+        /// </summary>
+        /// <param name="type"></param>
```
Comment (Seoyeon2Kim) [inline src/Tizen.NUI/src/public/Utility/Capture.cs:385]: Please add a description about param name even though it's very brief.
```suggestion
        /// <param name="type">DisposeTypes</param>
```

**Review by Seoyeon2Kim — COMMENTED** [review]
Please check my minor comments. :)

**Comment by dongsug-song** [comment]
@bshsqa 
Could you update the ACR link here?
as you know, this PR can not be merged before the ACR is accepted.

File: src/Tizen.NUI/src/public/Utility/Capture.cs [hunk 7]
```diff
@@ -333,12 +378,39 @@ public NativeImageSource GetNativeImageSource()
...
+            {
+                //Called by User
+                //Release your own managed resources here.
+                //You should release all of your own disposable objects here.
+            }
+
+            base.Dispose(type);
+        }
+
+        /// This will not be public opened.
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/Utility/Capture.cs:403]: Please add `EditorBrowsable(Never)` if don't want to show this.

File: src/Tizen.NUI/src/public/Utility/Capture.cs (hunk 1 above)
Comment (bshsqa) [reply src/Tizen.NUI/src/public/Utility/Capture.cs:85]: Thank you.

File: src/Tizen.NUI/src/public/Utility/Capture.cs (hunk 2 above)
Comment (bshsqa) [reply src/Tizen.NUI/src/public/Utility/Capture.cs:86]: Good thank you.

File: src/Tizen.NUI/src/public/Utility/Capture.cs (hunk 3 above)
Comment (bshsqa) [reply src/Tizen.NUI/src/public/Utility/Capture.cs:92]: Thank you.

File: src/Tizen.NUI/src/public/Utility/Capture.cs (hunk 4 above)
Comment (bshsqa) [reply src/Tizen.NUI/src/public/Utility/Capture.cs:119]: Thank you for your good check.

File: src/Tizen.NUI/src/public/Utility/Capture.cs (hunk 6 above)
Comment (bshsqa) [reply src/Tizen.NUI/src/public/Utility/Capture.cs:385]: Thank you.

File: src/Tizen.NUI/src/public/Utility/Capture.cs (hunk 7 above)
Comment (bshsqa) [reply src/Tizen.NUI/src/public/Utility/Capture.cs:403]: Thank you.

File: src/Tizen.NUI/src/public/Utility/Capture.cs (hunk 5 above)
Comment (bshsqa) [reply src/Tizen.NUI/src/public/Utility/Capture.cs:136]: Thank you x2

**Comment by dongsug-song** [comment]
@bshsqa 
could you share the ACR link?

and now I put the **\*\*NO MERGE\*\*** tag on this PR.
you can remove **\*\*NO MERGE\*\*** tag for yourself when this PR is **ACR accepted** and ready to be merged.
(**\*\*NO MERGE\*\*** tag has no restriction to submit PR or to edit this patch but just for notification and marking.
as you know, this PR can not be merged before the ACR is accepted.
thank you.

**Comment by pkosko** [comment]
There are several 'Github Actions' warnings in this ACR, could you please check them?

**Pull request closed without merging**